cs/src/Management/TunnelExtensions.cs

```diff
@@ -72,6 +72,64 @@ public static bool TryGetAccessToken(this Tunnel tunnel, string accessTokenScope
         return false;
     }
 
+    /// <summary>
+    /// Try to get an access token from <paramref name="tunnel"/> for <paramref name="accessTokenScope"/>.
+    /// </summary>
+    /// <remarks>
+    /// The tokens are searched in <c>Tunnel.AccessTokens</c> dictionary where each
+    /// key may be either a single scope or space-delimited list of scopes.
+    /// </remarks>
+    /// <param name="tunnel">The tunnel to get the access token from.</param>
+    /// <param name="accessTokenScope">Access token scope to get the token for.</param>
+    /// <param name="accessToken">If non-null and non-empty token is found, the token value. <c>null</c> if not found.</param>
+    /// <returns>
+    /// <c>true</c> if <paramref name="tunnel"/> has non-null and non-empty an access token for <paramref name="accessTokenScope"/>;
+    /// <c>false</c> if <paramref name="tunnel"/> has no access token for <paramref name="accessTokenScope"/> or the token is null or empty.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">If <paramref name="tunnel"/> or <paramref name="accessTokenScope"/> is null.</exception>
+    /// <exception cref="ArgumentException">If <paramref name="accessTokenScope"/> is empty.</exception>
+    public static bool TryGetAccessTokenV2(this TunnelV2 tunnel, string accessTokenScope, [NotNullWhen(true)] out string? accessToken)
```

Comment:
Can we just add an overload with the same name, `TryGetAccessToken` ? C# will use either the old one or the new one based on the type of `tunnel`.